### PR TITLE
feat: add useTimeout and Delay

### PR DESCRIPTION
**Pull request by Stefanwullems**
For https://exivity.atlassian.net/browse/EXVT-1573?atlOrigin=eyJpIjoiY2QzNThhZjNiZjc3NGQ5OGJlMzRhNzIyMzk5NDdiMTkiLCJwIjoiaiJ9

### CODE REVIEW

File: src/useTimeout/useTimeout.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { useEffect, useRef } from 'react'
```
Comment (Stefanwullems) [inline src/useTimeout/useTimeout.ts]: copy pasted from https://github.com/siddharthkp/use-timeout and tested

File: src/Delay/Delay.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+interface DelayProps {
+  wait: number
+  children: ReactNode
+}
+
+export const Delay: FC<DelayProps> = ({ wait, children }) => {
+  const [show, setShow] = useState(false)
+
+  const showChildren = useCallback(() => {
```
Comment (hongaar) [inline src/Delay/Delay.tsx]: I'm not sure if this optimisation makes sense tbh. We're not using the returned `showChildren` to optimise an expensive operation, so maybe the useCallback hook itself adds more overhead than we save (re-creating the function on each render).

Definitely, the code becomes less complex if we simplify this to

```js
const [show, setShow] = useState(false)

useTimeout(() => setShow(true), wait)
```

File: src/useTimeout/useTimeout.ts (hunk 1 above)
Comment (hongaar) [reply src/useTimeout/useTimeout.ts]: We do we not simply use this package here?

File: src/useTimeout/useTimeout.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+export function useTimeout (callback: Function, delay: number) {
+  const savedCallback = useRef<Function>()
+
+  useEffect(() => {
+    savedCallback.current = callback
+  }, [callback])
+
+  useEffect(() => {
+    if (delay !== null) {
```
Comment (hongaar) [inline src/useTimeout/useTimeout.ts]: This check may not be needed? (Was the original package also in TS?)

File: src/useTimeout/useTimeout.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { useEffect, useRef } from 'react'
+
+export function useTimeout (callback: Function, delay: number) {
+  const savedCallback = useRef<Function>()
```
Comment (hongaar) [inline src/useTimeout/useTimeout.ts]: TIL `useRef` can be useful outside the context of DOM nodes.
📚 https://reactjs.org/docs/hooks-reference.html#useref

File: src/Delay/Delay.tsx (hunk 2 above)
Comment (Stefanwullems) [reply src/Delay/Delay.tsx]: Good point

**Pull request merged**